edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -64,19 +63,39 @@ def test_has_permission(self, has_user_in_url):
 
 
 @ddt.ddt
-class IsJwtAuthenticatedTests(TestCase):
-    """ Tests for the IsJwtAuthenticated permission class. """
+class JwtRestrictedApplicationTests(TestCase):
+    """ Tests for the JwtRestrictedApplication and NotJwtRestrictedApplication permission classes. """
 
-    @ddt.data(JwtAuthentication, BaseJSONWebTokenAuthentication, SessionAuthentication, None)
-    def test_has_permission(self, authentication_class):
+    @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
+    @ddt.data(
+        *product(
+            (permissions.JwtRestrictedApplication, permissions.NotJwtRestrictedApplication),
+            (JwtAuthentication, BaseJSONWebTokenAuthentication, SessionAuthentication, None),
+            (True, False),
+            (True, False),
+        )
+    )
+    @ddt.unpack
+    def test_has_permission(self, permission_class, authentication_class, is_restricted, enforce_scopes, waffle_mock):
+        waffle_mock.return_value = enforce_scopes
         request = RequestFactory().get('/')
         request.successful_authenticator = authentication_class() if authentication_class else None
         request.user = factories.UserFactory()
-        request.auth = generate_jwt(request.user)
-        self.assertEqual(
-            permissions.IsJwtAuthenticated().has_permission(request, None),
-            issubclass(type(request.successful_authenticator), BaseJSONWebTokenAuthentication),
-        )
+        request.auth = generate_jwt(request.user, is_restricted=is_restricted)
+
+        is_jwt_auth_subclass = issubclass(type(request.successful_authenticator), BaseJSONWebTokenAuthentication)
+
+        has_permission = permission_class().has_permission(request, view=None)
+        if permission_class == permissions.JwtRestrictedApplication:
```

Comment:
If you used:
```
expected_restricted_permission = enforce_scopes and is_restricted and is_jwt_auth_subclass
```
Below, you could use `expected_restricted_permission` and `not expected_restricted_permission`.

Reply:
True. Updated.